x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -105,102 +60,252 @@ func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgWithdrawPosition, error) {
 	rand := sim.GetRand()
 	// get random pool
-	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// get random user address with the pool denoms
-	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
-	if !senderExists {
-		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	// positionCreator creates the position with pool denoms
+	positionCreator, tokens, lowerTick, upperTick, err := RandomPrepareCreatePositionFunc(sim, ctx, clPool, poolDenoms)
+	if err != nil {
+		return nil, err
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
-	if err != nil {
-		return nil, fmt.Errorf("position does not exist")
+	var positionIds []uint64
+	remainingToken0Amt := tokens[0].Amount
+	remainingToken1Amt := tokens[1].Amount
+	// create positions until the funds run out from positionCreator
+	for remainingToken0Amt.GT(sdk.ZeroInt()) && remainingToken1Amt.GT(sdk.ZeroInt()) {
+		//make sure the user always has tokens to create Position
+		randToken0Amt := sim.RandomAmount(remainingToken0Amt)
+		randToken1Amt := sim.RandomAmount(remainingToken1Amt)
+
+		positionId, _, _, _, _, err := k.CreatePosition(ctx, clPool.GetId(), positionCreator, randToken0Amt, randToken1Amt, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick, upperTick)
+		if err != nil {
+			return nil, err
+		}
+
+		remainingToken0Amt = remainingToken0Amt.Sub(randToken0Amt)
+		remainingToken1Amt = remainingToken1Amt.Sub(randToken1Amt)
+
+		positionIds = append(positionIds, positionId)
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	randPositionId := positionIds[rand.Intn(len(positionIds))]
+	position, err := k.GetPosition(ctx, randPositionId)
+	if err != nil {
+		return nil, err
```

Comment:
This is how I see this working best:

1. Randomly select a CL pool
2. Return all CL position IDs that are currently in that pool
3. Randomly select one of these CL positions
4. Withdraw a random amount from this position

Creating a position every time we withdraw feels like we could be missing out on some important edge cases

Reply:
I see, i originally thought we wanted to test the "entire flow", for instance; 
createPool -> createPosition -> performSwaps -> collect fees 
while randomizing the createPool parameters, selecting a random position from that position, performing lots of swaps and collecting random fees for random positions

Reply:
but understandably your methods will create more randomness, which also makes sense